Deletebutton

Liliya-V opened the pull request.

Liliya-V committed:
created editformdb.php, wrote a code for inserting editeddata into a DB

Liliya-V committed:
changes a title

Liliya-V committed:
Merge branch 'master' into editformdb

Liliya-V committed:
added pictures

Liliya-V committed:
merged master branch in.
set an id to edit form sochanged a select insert function to update one, set an id to edit a correct position

Liliya-V committed:
set a substraction of first 7 characters from an 'image' field of an edit form, so the path to the image would not have a doubled directory

Liliya-V committed:
created a deletebutton page

Liliya-V committed:
Wrote a code for running a deleting query, passed a href to the deletebutton page into the admin panel page so the button would work.

mporam commented on adminPanel/deletebutton.php:
```diff
@@ -0,0 +1,16 @@
+if (!empty($_GET['id'])) {
+    $id = $_GET['id'];
+}
+$sqlDelete = 'DELETE FROM `projects` WHERE `id`= ' . $id . ';';
```
please no sql injections

mporam commented on adminPanel/deletebutton.php:
```diff
@@ -0,0 +1,16 @@
+require('functions.php');
+$db = new PDO("mysql:host=192.168.20.20;dbname=PortfolioProjects", 'root', '');
+$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+if (!empty($_GET['id'])) {
```
what happens if the id is empty? this page will error

Liliya-V committed:
removed an sql injection vulnerability,
set the if statement for GET id

Liliya-V committed:
changed pictures sise,
added a 'go to your portfolio" button,
removed a header to a different page from a delet button so it stays on a same page

mporam commented on adminPanel/deletebutton.php:
```diff
@@ -0,0 +1,21 @@
+        echo '<br><a href="adminPanel.php">Try again</a><br>';
+        echo '<br><a href="../portfolio.php">Go to your portfolio.</a><br>';
+    } else {
+        header('location:adminPanel.php?objectdeleted');
```
you dont need the `?objectdeleted` on the url, it will be obvious on the admin panel that it ha been deleted

Liliya-V committed:
removed a ?objectdeleted from the URL

Liliya-V committed:
Merge branch 'master' into deletebutton
